doc/sonic-sub-port-intf-hld.md

```diff
@@ -0,0 +1,475 @@
+
+Multiple L3 sub port interfaces, each characterized by a VLAN id in the 802.1q tag, can be created on a physical port or a port channel.
+Sub port interfaces attaching to the same physical port or port channel can interface to different VRFs, though they share the same VLAN id space and must have different VLAN ids.
+Sub port interfaces attaching to different physical ports or port channels can use the same VLAN id, even when they interface to the same VRF.
+However, there is no L2 bridging between these sub port interfaces; each sub port interface is considered to stay in a separate bridge domain.
+
+# Acronyms
+| Acronym                  | Description                                |
+|--------------------------|--------------------------------------------|
+| VRF                      | Virtual route forward                      |
+| RIF                      | Router interface                           |
+
+# 1 Requirements
+
+Manage the life cycle of a sub port interface created on a physical port or a port channel and used as a router interface to a VRF:
+* Creation with the specified dot1q vlan id encapsulation and optional mtu
+* Runtime mtu change
+* Removal
+
+A sub port interface shall support the following features:
+* L3 forwarding (both unicast and multicast)
+* BGP
+* ARP
+* VRF
+* RIF counters
+* QoS setting inherited from parent physical port or port channel
+* Per sub port interface mtu config
+* mtu inheritance from parent physical port or port channel if mtu is not specified on a sub port interface in the configuration file.
+
+# 2 Schema design
+
+We introduce a new table "DOT1Q_INTERFACE" in the CONFIG_DB to host the attributes of a sub port interface.
+For APPL_DB and STATE_DB, we do not introduce new tables for sub port interfaces, but reuse existing tables to host sub port interface keys.
+
+## 2.1 Configuration
+### 2.1.1 config_db.json
+```
+"DOT1Q_INTERFACE": {
+    "{{ port_name }}.{{ vlan_id }}": {
+        "mtu" : "{{ mtu_size }}"
+    },
+    "{{ port_name }}.{{ vlan_id }}|{{ ip_prefix }}": {}
+},
+
+"VLAN": {
+    "Vlan{{ vlan_id }}": {
+        "vlanid" : "{{ vlan_id }}"
+    }
+},
+```
+A key in the DOT1Q_INTERFACE table is the name of a sub port, which consists of two sections delimited by a "." (symbol dot). The section before the dot is the name of the parent physical port or port channel. The section after the dot is the dot1q encapsulation vlan id.
+
+mtu of a sub port interface should be no greater than that of the parent physical port or port channel. In the case field "mtu" is absent in the config_db.json file, a sub port interface inherits the mtu from its parent physical port or port channel.
+
+Example configuration:
+```
+"DOT1Q_INTERFACE": {
+    "Ethernet64.10": {
+        "mtu" : "9100"
+    },
+    "Ethernet64.10|192.168.0.1/21": {}
+},
+
+"VLAN": {
+    "Vlan10": {
+        "vlanid" : "10"
+    }
+},
+```
+
+### 2.1.2 CONFIG_DB
```

Comment:
Is there an concept of port alias in this usage? For example Ethernet64.10 -> etp16.10

Reply:
Not in CONFIG_DB. But in CLI, we support interface config using port alias, and the internally CLI will map port alias to default port name to set the CONFIG_DB.